lib/nostrum/struct/snowflake.ex

```diff
@@ -115,7 +151,7 @@ defmodule Nostrum.Struct.Snowflake do
   def creation_time(snowflake) when is_snowflake(snowflake) do
     use Bitwise
 
-    time_elapsed_ms = (snowflake >>> 22) + 1_420_070_400_000
+    time_elapsed_ms = (snowflake >>> 22) + @discord_epoch
 
     DateTime.from_unix!(time_elapsed_ms, :milliseconds)
```

Comment:
Looking at this now, the spec doesn't have `no_return` listed as a possible return value.

Reply:
I think we talked about this before, but this function actually cannot fail; pretty much impossible since the `is_snowflake/1` guard prevents inputting numbers that can make the function raise an error.

Reply:
Dialyzer won't agree with you I think 🙃

Reply:
I've ran dialyzer a couple of times. Not warning us about anything so far. If you can run it on your machine and see if a warning pops up, we'll definitely change it. And if a warning does pop up, we'll need to explain to the user that the function can't fail and that the no_return is to make dialyzer be quiet. :\

Reply:
[Well...](https://github.com/elixir-lang/elixir/blob/v1.6.6/lib/elixir/lib/calendar/datetime.ex#L165)

Also, we're doing checks on our own, but if something changes behind the scenes we'll want to pass any errors through. Under certain circumstances it probably can fail for w/e reason, even though with our code it's very unlikely. I think it's important to have that as part of the spec because it's not under our direct control.

Reply:
Fixed in 9acfcbe.